CamDetection.as

```diff
@@ -0,0 +1,146 @@
+		g_activeCam = ActiveCam::Backwards;
+	} else if (camModelNod !is null) {
+		auto ty = Reflection::TypeOf(camModelNod);
+		switch (ty.ID) {
+			// CPlugVehicleCameraRace2Model
+			case 0x90f6000: {
+				g_activeCam = TestCam1Alt(camModelNod) ? ActiveCam::Cam1Alt : ActiveCam::Cam1;
+				break;
+			}
+			// CPlugVehicleCameraRace3Model
+			case 0x90ef000: {
+				g_activeCam = TestCam2Alt(camModelNod) ? ActiveCam::Cam2Alt : ActiveCam::Cam2;
+				break;
+			}
+			// CPlugVehicleCameraInternalModel
+			case 0x90f7000: {
+				g_activeCam = TestCam3Alt(camModelNod) ? ActiveCam::Cam3Alt : ActiveCam::Cam3;
+				break;
+			}
+			default: {
+				trace('1 Got cam of unknown type: ' + ty.ID + ', ' + ty.Name);
+				UI::ShowNotification('1 Got cam of unknown type: ' + ty.ID + ', ' + ty.Name);
+				g_activeCam = ActiveCam::Other;
+			};
+		}
+	} else if (camControlNod !is null) {
+		auto ty = Reflection::TypeOf(camControlNod);
+		switch (ty.ID) {
+			// CGameControlCameraFree
+			case 0x306d000: {
+				g_activeCam = ActiveCam::FreeCam;
+				break;
+			}
+			// CGameControlCameraEditorOrbital
+			case 0x3125000: {
+				g_activeCam = ActiveCam::EditorOrbital;
+				break;
+			}
+			// CGameControlCameraOrbital3d
+			case 0x306e000: {
+				g_activeCam = ActiveCam::Orbital3d;
+				break;
+			}
+			// CGameControlCameraHelico
+			case 0x32f0000: {
+				g_activeCam = ActiveCam::Helico;
+				break;
+			}
+			// CGameControlCameraHmdExternal
+			case 0x3277000: {
+				g_activeCam = ActiveCam::HmdExternal;
+				break;
+			}
+			// CGameControlCameraThirdPerson
+			case 0x3042000: {
+				g_activeCam = ActiveCam::ThirdPerson;
+				break;
+			}
+			// CGameControlCameraFirstPerson
+			case 0x3041000: {
+				g_activeCam = ActiveCam::FirstPerson;
+				break;
+			}
+			// CGameControlCameraTarget
+			case 0x3072000: {
+				g_activeCam = ActiveCam::Target;
+				break;
+			}
+			// CGameControlCameraTrackManiaRace
+			case 0x313d000: {
+				g_activeCam = ActiveCam::Cam0;
+				break;
+			}
+			// CGameControlCamera
+			case 0x306b000: {
+				// We probably won't ever end up here, but just in case.
+				g_activeCam = ActiveCam::Other;
+				break;
+			}
+			// CGameControlCameraTrackManiaRace2
+			// 0x31cf000
+			// CGameControlCameraTrackManiaRace3
+			// 0x31ce000
+			// CGameControlCameraVehicleInternal
+			// 0x3189000
+			default: {
+				trace('2 Got cam of unknown type: ' + ty.ID + ', ' + ty.Name);
+				UI::ShowNotification('2 Got cam of unknown type: ' + ty.ID + ', ' + ty.Name);
+				g_activeCam = ActiveCam::Other;
+			};
+		}
+	} else {
+		// initalizing editor, when in mediatracker
+		g_activeCam = ActiveCam::Loading;
+	}
+}
+
+bool TestCam1Alt(CMwNod@ camNod)
+{
+	// alt 00 00 40 40, main: 00 00 a0 40 -- constant after restart
+	return Dev::GetOffsetUint32(camNod, 0x38) == 0x40400000;
+}
+
+bool TestCam2Alt(CMwNod@ camNod)
+{
+	// alt: 00 00 8C 42, main: 00 00 96 42 -- constant after game restart
+	return Dev::GetOffsetUint32(camNod, 0x38) == 0x428c0000;
+}
```

Comment:
These constants feel pretty arbitrary. What do they mean? I'm not sure I'm comfortable adding these without fully understanding them.

Reply:
Yeah, I don't know what they do, just that they're different.

Another way I found to detect alt cam is via .GameTerminals[0] offset 0x30. This can also be set to 0 or 1 to set the alt cam on/off. 

I opted for the arbitrary test method b/c everything was in one place, but the game terminal method is much simpler and clearer. (Probably good for maintenance, too)

Reply:
Actually, the game terminal has more info about cam 1/2/3/7, however, that has less information than the method in the PR (e.g., the PR detects EditorOrbital, but the game terminal method wont).

Reply:
Found a better way to test for alt cams -- see latest version. There are some uint8 flags in the SCamSys (0x1 for std, 0x2 for alt).